### PR TITLE
Correctly parse json schema union types

### PR DESCRIPTION
**Pull request by sherifnada**
## What
A valid JSONSchema type declaration can be: 

```
"type": ["null", "integer", "string"]
```

which means that the type can be either null, integer, or string. When Airbyte processes such a field, we always want to treat it as a string (at least for the moment). We do not handle this type gracefully in normalization or in converting catalogs to `Schema` objects. This causes connectors like Shopify to break when writing normalized data into a destination. 

This PR fixes the issue by treating this column as a string when normalizing and converting a catalog to a `Schema` object. 

## Pre-merge Checklist
- [ ] *Publish new normalization image*

## Recommended reading order
**Normalization changes:** 
1. transform.py
2. `TestDestination.java`
3. *.json

**Catalog conversion**
1. `AirbyteProtocolConverters.java`
2. AirbyteProtocolConvertersTest.java

### CODE REVIEW

File: airbyte-integrations/bases/base-normalization/normalization/transform_catalog/transform.py [hunk 1]
```diff
@@ -145,8 +145,12 @@ def read_json_catalog(input_path: str) -> dict:
     return json.loads(contents)
 
 
+def is_union_type(property_type) -> bool:
+    return type(property_type) is list and len(property_type) > 2
```
Comment (ChristopheDuong) [inline airbyte-integrations/bases/base-normalization/normalization/transform_catalog/transform.py:149]: What if we get a union type non-nullable? 
`"type": ["integer", "string"]`

File: airbyte-workers/src/main/java/io/airbyte/workers/normalization/DefaultNormalizationRunner.java [hunk 2]
```diff
@@ -43,7 +43,7 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(DefaultNormalizationRunner.class);
 
-  public static final String NORMALIZATION_IMAGE_NAME = "airbyte/normalization:0.1.10";
+  public static final String NORMALIZATION_IMAGE_NAME = "airbyte/normalization:dev";
```
Comment (cgardens) [inline airbyte-workers/src/main/java/io/airbyte/workers/normalization/DefaultNormalizationRunner.java:46]: don't forget to change this back.

**Review by cgardens — COMMENTED** [review]
Is there a reasonable approach where we solve this in the shopify source? And then we say we only support a subset of the JsonSchema spec (we support single data type or single data type + null)? At least for now.

Here are some of the tradeoffs I am thinking about:
* this is a hack and leads to unexpected behavior
* we do already have some hacks around types and JsonSchema though
* we may need to make our own type system more robust to just support unions, but that's probably not a change we can do now.
* It seems like if we can just assert that we only support a subset of the JsonSchema spec in this case then at least that's straightforward. 
* The behavior in this PR is kinda hard to think about. What are the risk areas that this could cause something downstream to break? Or is it relatively safe because String is generally just our escape hatch?

Sorry, I am failing at my job as a reviewer right now, by not providing a go / no go. But I would need some more time to think about this.

File: airbyte-integrations/bases/base-normalization/normalization/transform_catalog/transform.py (hunk 1 above)
Comment (sherifnada) [reply airbyte-integrations/bases/base-normalization/normalization/transform_catalog/transform.py:149]: good point, will change to handle non-nullables

**Comment by sherifnada** [comment]
@cgardens I think this is "safe" in that string should always be a viable escape hatch. The choice now is whether we want to attempt to support unions right now or patch the shopify source to prevent this problem. Some holes I see with this handling of union types are: 
1. String isn't necessarily the "best" option for handling any union types -- it depends on the types in the union e.g `[integer, double]` should not be converted to string
2. Ultimately i think we should let each destination handle coverting from JSONSchema (or whatever subset of it we allow) to its type system e.g: mysql might not have the ability to handle arrays, but Postgres might. In this case we don't want Airbyte core dictating what the type is; we want to surface that as-is and let the destination handle it. 

I think this is convincing me that we shouldn't make changes to the type system as part of fixing the problem with Shopify or other sources. But this _is_ a timebomb for any community-contributed singer taps. Let's call a type system fix a fast follow? :) 

PS: expecting go/no-go is nice but overrated. Problems are [elephants, and partial feedback can help us see the whole. ](https://en.wikipedia.org/wiki/Blind_men_and_an_elephant)

**Comment by sherifnada** [comment]
@cgardens another point here is that if we are saying we don't handle union types right now, then we should proactively verify that in source tests, none of the catalogs generated by a source contain union types. Either that or we merge this PR to use strings as a stop-gap. What I don't want to be doing is reactively finding out about normalization issues because a source sneaked a union type under the radar.

**Comment by jrhizor** [comment]
I like the idea of standard test enforcement.

**Pull request closed without merging**